### ISSUE TITLE
Renew Certificate freeze XCA

### ISSUE THREAD

**Issue by scasrl** [chris2511/xca#493]
When I renew a certificate if I tick the Revoke certificate AND Replace old certificate boxes, I think why it can't update the DB

**Comment by C4rb0n** [comment]
I can confirm (using latest XCA v2.5.0).

When renewing a certificate, if you select only "Replace old certificate" and/or "Keep serial number," everything is fine.  However, if you also select "Revoke old certificate," the UI hangs indefinitely and the application must be forcefully terminated.

The certificate does get successfully created, as evident when relaunching and logging back in.

**Comment by jmahmens** [comment]
I can confirm also this error but I see an Error message:
```
Der folgende Fehler ist aufgetreten:
(8pki_x509[]:switch0XXXX)
error:0300009C:digital envelope routines::unsupported algorithm
error:068C0100:asn1 encoding routines::malloc failure
error:068C0100:asn1 encoding routines::malloc failure

(C:\Users\chris\build\xca-2.5.0\lib\pki_x509.cpp:582)
```

**Comment by lbrunkho** [comment]
On my system, this results in SIGABRT killing xca. Closing RHEL auto-generated [issue](https://bugzilla.redhat.com/show_bug.cgi?id=2259477). Some additional attachments / crash dumps can be found here if needed.

**Comment by mrjester888** [comment]
Having the same issue.

**Comment by chris2511** [comment]
Fixed with #507

You are invited to verify the fix with the builds from https://github.com/chris2511/xca/actions/runs/7806292181
